runtime/runtime_events.c

```diff
@@ -716,7 +717,7 @@ CAMLprim value caml_runtime_events_user_register(value event_name,
   Field(event, 3) = event_tag;
 
 
-  caml_plat_lock_blocking(&user_events_lock);
+  caml_plat_lock_non_blocking(&user_events_lock);
```

Comment:
In this critical section, we call `caml_alloc_small` and therefore we cannot use `caml_plat_lock_blocking`